### PR TITLE
[Breaking change] Remove Bang Operator

### LINKED ISSUE #288: Improve the gem's public api by removing the bang operator in some methods
Original discussion https://github.com/meilisearch/meilisearch-ruby/pull/273/files#r779759981

Today we use `!`(bang) operator in some particular cases eg. `create_index!`, `add_documents!`, these methods have something in common, *they wait for the task completion*.

After v0.25.0 index creations are totally async, so they respond with a task object not an index anymore. Anyway, what is the real problem here is the usage of bang makes a lot of difference and the typing structure is close to the other methods.

You could type `create_index` and get a `task` as a response, or  
You could type `create_index!` and get an `index` as a response.

The proposal here is to remove this syntax in order to provide something more clear to prevent such mistakes like:

`create_index('movies', sync: true)`, `create_index('movies', wait: true)`, `create_index('movies', wait_for_completion: true)`, in a way that if we don’t pass such param we know that it will respond with default `task` object.

### PR DESCRIPTION
**Pull request by yyamanoi1222**
# Pull Request

## Related issue
Fixes #288

## What does this PR do?
- remove bang operator in method name and add wait option
- some breaking changes are included in the way the arguments are given

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: lib/meilisearch/client.rb [hunk 1]
```diff
@@ -31,17 +31,14 @@ def indexes(options = {})
 
     # Usage:
     # client.create_index('indexUID')
-    # client.create_index('indexUID', primaryKey: 'id')
-    def create_index(index_uid, options = {})
+    # client.create_index('indexUID', { primaryKey: 'id' })
+    # client.create_index('indexUID', { primaryKey: 'id' }, wait: true)
+    def create_index(index_uid, options = {}, wait: false)
```
Comment (yyamanoi1222) [inline lib/meilisearch/client.rb:36]: This change has breaking change about passing arguments.
I think that passing keyword arguments to argument of hash is possible but deprecated ruby3 and above.
https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/

Comment (yyamanoi1222) [reply lib/meilisearch/client.rb]: Changed to keep hash definition for compatibility

**Comment by brunoocasali** [comment]
Hi @yyamanoi1222 thanks a lot for your contribution, I'll run the CI for your PR, and I will check it as soon as possible!

I'm glad to see you here!

**Comment by brunoocasali** [comment]
Hey @yyamanoi1222 I didn't forget you, I'm just having a hard moment to find the time to review your PR, I'll do it ASAP!

**Comment by brunoocasali** [comment]
Hey @yyamanoi1222, I want to thank you for the hard work you did here. Unfortunately, the solution proposed by @ellnix fits the library better, and I think it will be more beneficial for the consumers.

Thanks again for your time, I will mention your name in the next release ;) 

See ya!

**Pull request closed without merging**